packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -56,73 +64,165 @@ export class Relayer {
     for (const l1Token of Object.keys(relayableDeposits)) {
       this.logger.debug({ at: "Relayer", message: "Processing relayable deposits for L1Token", l1Token });
       for (const relayableDeposit of relayableDeposits[l1Token]) {
-        const botL1TokenBalance = await this.fetchBotL1TokenBalance(l1Token);
-        const relayType = this.shouldRelay(relayableDeposit, botL1TokenBalance);
-        switch (relayType) {
-          case RelayType.IgnoreRelay:
-            this.logger.info({
-              at: "Relayer",
+        const realizedLpFeePct = await this.l1Client.calculateRealizedLpFeePctForDeposit(relayableDeposit.deposit);
+        const desiredRelayType = await this.shouldRelay(
```

Comment:
I think it might make sense to rename this variable to `shouldRelay` and rename the `RelayType` enum to `ShouldRelay{No, Slow, Instant}`